### PR TITLE
Get compliance reports for a deployment

### PR DESCRIPTION
**Pull request by nikitajoshi1**
# Description
Get compliance report for a Deployment and related tests.

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
**Test Result:**
gerall
withFilter

### CODE REVIEW

File: service.go [hunk 1]
```diff
@@ -653,3 +653,90 @@ func (gc *GatewayClient) DeleteService(serviceID, serversInInventory, serversMan
...
+	}
+	if httpResp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Couldn't find compliance report for given deployment")
+	}
+
+	var complianceReports []types.ComplianceReport
+	responseString, _ := extractString(httpResp)
+	parseError := json.Unmarshal([]byte(responseString), &complianceReports)
+	if parseError != nil {
+		return nil, fmt.Errorf("Error While Parsing Response Data For Compliance Report: %s", parseError)
```
Comment (AkshaySainiDell) [inline service.go:695]: Please change the error message to sentence case.

File: service.go [hunk 2]
```diff
@@ -653,3 +653,90 @@ func (gc *GatewayClient) DeleteService(serviceID, serversInInventory, serversMan
...
+
+	return complianceReports, nil
+}
+
+// GetServiceComplianceDetailsByFilter retrieves service compliance details based on a filter and value.
+func (gc *GatewayClient) GetServiceComplianceDetailsByFilter(deploymentID, filter, value string) ([]types.ComplianceReport, error) {
+	defer TimeSpent("GetServiceComplianceDetailsByFilter", time.Now())
+
+	complianceReports, err := gc.GetServiceComplianceDetails(deploymentID)
+	if err != nil && len(complianceReports) == 0 {
```
Comment (AkshaySainiDell) [inline service.go:709]: I think an OR check makes more sense here. Please check and update if required.
```suggestion
	if err != nil || len(complianceReports) == 0 {
```

File: types/v1/serviceTypes.go
Comment (AkshaySainiDell) [inline types/v1/serviceTypes.go:1]: Copyright info is missing.

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

**Comment by AkshaySainiDell** [comment]
Please add the github issue link in PR description.

File: service.go (hunk 1 above)
Comment (nikitajoshi1) [reply service.go]: @AkshaySainiDell  Updated the error message.

File: service.go (hunk 2 above)
Comment (nikitajoshi1) [reply service.go]: That seems reasonable . Updated.

**Comment by nikitajoshi1** [comment]
> Please add the github issue link in PR description.

@AkshaySainiDell  Created a github issue and added here.

**Review by doriac11 — APPROVED** [review]

**Review by AkshaySainiDell — APPROVED** [review]
LGTM

**Pull request merged**